compiler/rustc_typeck/src/check/method/confirm.rs

```diff
@@ -161,14 +161,15 @@ impl<'a, 'tcx> ConfirmContext<'a, 'tcx> {
         assert_eq!(n, pick.autoderefs);
 
         let mut adjustments = self.adjust_steps(&autoderef);
+        let mut target = self.structurally_resolved_type(autoderef.span(), ty);
 
-        let mut target =
-            self.structurally_resolved_type(autoderef.span(), autoderef.final_ty(false));
```

Comment:
afaict `autoderef.final_ty` is identical to the type returned by `autoderef.nth`, so this was a bit redundant